File: app/src/main/java/com/dev/ome/corentin/xkcdapp/ComicCardActivity.java
package com.dev.ome.corentin.xkcdapp;

import android.content.Intent;
import android.graphics.BitmapFactory;
import android.os.Bundle;
import android.support.v4.view.MenuItemCompat;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.ShareActionProvider;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.ImageView;
import android.widget.TextView;

import java.io.ByteArrayInputStream;


public class ComicCardActivity extends AppCompatActivity {

    private ImageView imageView;
    private TextView safeTitle;
    private TextView year;
    private TextView transcript;
    private TextView url;
    private byte[] img;
    private String urlString;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_comic_card);

        this.imageView = findViewById(R.id.comicView);
        this.safeTitle = findViewById(R.id.safeTitle);
        this.year = findViewById(R.id.year);
        this.transcript = findViewById(R.id.transcript);
        this.url = findViewById(R.id.url);

        this.urlString = getIntent().getStringExtra("url");
        this.img = getIntent().getByteArrayExtra("image");

        this.imageView.setImageBitmap(BitmapFactory.decodeStream(new ByteArrayInputStream(img)));
        this.safeTitle.setText(getIntent().getStringExtra("title"));
        this.year.setText(getIntent().getStringExtra("year"));
        this.transcript.setText(getIntent().getStringExtra("transcript"));
        this.url.setText(this.urlString);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_comic_card, menu);

        MenuItem shareItem = menu.findItem(R.id.menu_item_share);
        ShareActionProvider myShareActionProvider = (ShareActionProvider) MenuItemCompat.getActionProvider(shareItem);

        Intent intent = new Intent(Intent.ACTION_SEND)
        .setType("text/plain")
        .putExtra(Intent.EXTRA_TEXT, this.urlString);

        myShareActionProvider.setShareIntent(intent);

        return true;
    }

}

File: app/src/main/java/com/dev/ome/corentin/xkcdapp/ComicCardActivity.kt
package com.dev.ome.corentin.xkcdapp

import android.content.Intent
import android.graphics.BitmapFactory
import android.os.Bundle
import android.support.v4.view.MenuItemCompat
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.ShareActionProvider
import android.view.Menu
import android.view.MenuItem
import android.widget.ImageView
import android.widget.TextView

import java.io.ByteArrayInputStream


class ComicCardActivity : AppCompatActivity() {

    private var imageView: ImageView? = null
    private var safeTitle: TextView? = null
    private var year: TextView? = null
    private var transcript: TextView? = null
    private var url: TextView? = null
    private var img: ByteArray? = null
    private var urlString: String? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_comic_card)

        this.imageView = findViewById(R.id.comicView)
        this.safeTitle = findViewById(R.id.safeTitle)
        this.year = findViewById(R.id.year)
        this.transcript = findViewById(R.id.transcript)
        this.url = findViewById(R.id.url)

        this.urlString = intent.getStringExtra("url")
        this.img = intent.getByteArrayExtra("image")

        this.imageView!!.setImageBitmap(BitmapFactory.decodeStream(ByteArrayInputStream(img)))
        this.safeTitle!!.text = intent.getStringExtra("title")
        this.year!!.text = intent.getStringExtra("year")
        this.transcript!!.text = intent.getStringExtra("transcript")
        this.url!!.text = this.urlString
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu_comic_card, menu)

        val shareItem = menu.findItem(R.id.menu_item_share)
        val myShareActionProvider = MenuItemCompat.getActionProvider(shareItem) as ShareActionProvider

        val intent = Intent(Intent.ACTION_SEND)
                .setType("text/plain")
                .putExtra(Intent.EXTRA_TEXT, this.urlString)

        myShareActionProvider.setShareIntent(intent)

        return true
    }

}
